tethys_apps/models.py

```diff
@@ -229,9 +236,131 @@ def get_value(self, *args, **kwargs):
+                                app_specific_settings = secret_app_settings[self.tethys_app.package]['custom_settings_salt_strings']
+                                if self.name in app_specific_settings:
+                                    app_custom_setting_salt_string = app_specific_settings[self.name]
+                                    if app_custom_setting_salt_string != '':
+                                        signer = Signer(salt=app_custom_setting_salt_string)
+                                    
+                        self.value = signer.sign_object(self.value)
+
+                    else:
+                        log.info(
+                            "There is not a an apps portion in the secrets.yml, please create one."
+                        )
+                        self.value = signer.sign_object(self.value)
+        
+    def get_value(self):
+        """
+        Get the value
+        """
+        if self.value == "" or self.value is None:
+            if self.required:
+                raise TethysAppSettingNotAssigned(
+                    f'The required setting "{self.name}" for app "{self.tethys_app.package}":'
+                    f"has not been assigned."
+                )
+
+            # None is a valid value to return in the case the value has not been set for this setting type
+            return None
+
+        TETHYS_HOME = get_tethys_home_dir()
+        signer = Signer()
+        secret_unsigned = ''
+
+        try:
+            if not os.path.exists(os.path.join(TETHYS_HOME, "secrets.yml")):
+                secret_unsigned = signer.unsign_object(f'{self.value}')
+            else:  
+                with open(os.path.join(TETHYS_HOME, "secrets.yml")) as secret_yaml:
+                    secrets_app_settings = yaml.safe_load(secret_yaml).get("secrets", {}) or {}
+                    if bool(secrets_app_settings):
+                        if 'custom_settings_salt_strings' in secrets_app_settings[self.tethys_app.package]:
+                            app_specific_settings = secrets_app_settings[self.tethys_app.package]['custom_settings_salt_strings']
+                            if self.name in app_specific_settings:
+                                app_custom_setting_salt_string = app_specific_settings[self.name]
+                                if app_custom_setting_salt_string != '':
+                                    signer = Signer(salt=app_custom_setting_salt_string)
+  
+                    secret_unsigned = signer.unsign_object(f'{self.value}')
+                
+        except signing.BadSignature:
+
+            raise TethysAppSettingNotAssigned(
+                f'The salt string for the setting {self.name} has been changed or lost, please enter the secret custom settings in the application settings again.'
+            )
+
+        return secret_unsigned
+
+
+class CustomSimpleSetting(CustomSetting):
```

Comment:
There’s an issue with calling this “CustomSimpleSetting” b/c it causes a breaking change. The “CustomSetting” has effectively been renamed to “CustomSimpleSetting” b/c existing apps with “CustomSettings” will need to rename them to “CustomSimpleSetting” to still function the same. To avoid the breaking change I suggest we make the following name changes:

- [ ] Rename the “CustomSetting” model to “CustomSettingBase”
- [ ] Rename the “CustomSimpleSetting” to “CustomSetting”

This should make it so that it will work with existing apps.

Reply:
- [ X ]  Rename the “CustomSetting” model to “CustomSettingBase”
- [ X  ]  Rename the “CustomSimpleSetting” to “CustomSetting”

Reply:
asdasd